Mathlib/NumberTheory/Transcendental/LindemannWeierstrass.lean

```diff
@@ -0,0 +1,1441 @@
+#align exp_polynomial_approx exp_polynomial_approx
+
+namespace Aux
+
+variable (p : ℚ[X])
+
+abbrev K' : IntermediateField ℚ ℂ :=
+  IntermediateField.adjoin ℚ (p.rootSet ℂ)
+set_option linter.uppercaseLean3 false in
+#align aux.K' Aux.K'
+
+instance K'.isSplittingField : IsSplittingField ℚ (K' p) p :=
+  IntermediateField.adjoin_rootSet_isSplittingField (IsAlgClosed.splits_codomain p)
+set_option linter.uppercaseLean3 false in
+#align aux.K'.is_splitting_field Aux.K'.isSplittingField
+
+abbrev K : Type _ :=
+  p.SplittingField
+set_option linter.uppercaseLean3 false in
+#align aux.K Aux.K
+
+instance : CharZero (K p) :=
+  charZero_of_injective_algebraMap (algebraMap ℚ (K p)).injective
+
+instance : IsGalois ℚ (K p) where
+
+abbrev Lift : K' p ≃ₐ[ℚ] K p :=
+  IsSplittingField.algEquiv (K' p) p
+set_option linter.uppercaseLean3 false in
+#align aux.Lift Aux.Lift
+
+instance algebraKℂ : Algebra (K p) ℂ :=
+  ((K' p).val.comp (Lift p).symm.toAlgHom).toRingHom.toAlgebra
+set_option linter.uppercaseLean3 false in
+#align aux.algebra_K_ℂ Aux.algebraKℂ
+
+instance : Algebra ℚ (K p) :=
+  inferInstance
+
+instance : SMul ℚ (K p) :=
+  Algebra.toSMul
+
+instance cache_ℚ_K_ℂ : IsScalarTower ℚ (K p) ℂ :=
+  inferInstance
+set_option linter.uppercaseLean3 false in
+#align aux.cache_ℚ_K_ℂ Aux.cache_ℚ_K_ℂ
+
+instance cache_ℤ_K_ℂ : IsScalarTower ℤ (K p) ℂ :=
+  inferInstance
+set_option linter.uppercaseLean3 false in
+#align aux.cache_ℤ_K_ℂ Aux.cache_ℤ_K_ℂ
+
+end Aux
+
+namespace Quot
+
+@[reducible] --, elab_as_elim]
+protected def liftFinsupp {α : Type _} {r : α → α → Prop} {β : Type _} [Zero β] (f : α →₀ β)
+    (h : ∀ a b, r a b → f a = f b) : Quot r →₀ β := by
+  refine' ⟨image (mk r) f.support, Quot.lift f h, fun a => ⟨_, _⟩⟩
+  · rw [mem_image]; rintro ⟨b, hb, rfl⟩; exact Finsupp.mem_support_iff.mp hb
+  · induction a using Quot.ind
+    rw [lift_mk _ h]; refine' fun hb => mem_image_of_mem _ (Finsupp.mem_support_iff.mpr hb)
+#align quot.lift_finsupp Quot.liftFinsupp
+
+theorem liftFinsupp_mk {α : Type _} {r : α → α → Prop} {γ : Type _} [Zero γ] (f : α →₀ γ)
+    (h : ∀ a₁ a₂, r a₁ a₂ → f a₁ = f a₂) (a : α) : Quot.liftFinsupp f h (Quot.mk r a) = f a :=
+  rfl
+#align quot.lift_finsupp_mk Quot.liftFinsupp_mk
+
+end Quot
+
+namespace Quotient
+
+@[reducible] --, elab_as_elim]
+protected def liftFinsupp {α : Type _} {β : Type _} [s : Setoid α] [Zero β] (f : α →₀ β) :
+    (∀ a b, a ≈ b → f a = f b) → Quotient s →₀ β :=
+  Quot.liftFinsupp f
+#align quotient.lift_finsupp Quotient.liftFinsupp
+
+@[simp]
+theorem liftFinsupp_mk' {α : Type _} {β : Type _} [Setoid α] [Zero β] (f : α →₀ β)
+    (h : ∀ a b : α, a ≈ b → f a = f b) (x : α) : Quotient.liftFinsupp f h (Quotient.mk' x) = f x :=
+  rfl
+#align quotient.lift_finsupp_mk Quotient.liftFinsupp_mk'
+
+end Quotient
+
+section
+
+variable (s : Finset ℂ)
+
+namespace Transcendental -- Conflict with Mathlib.NumberTheory.Dioph
+abbrev Poly : ℚ[X] :=
+  ∏ x in s, minpoly ℚ x
```

Comment:
I'm wondering whether i would be easier to take the [normalClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/NormalClosure.html#normalClosure) of `IntermediateField.adjoin ℚ s` in `ℂ`, or we can do it in one step by adjoining all conjugates (same minpoly) of elements in `s`. Or maybe we can directly work with automorphisms of ℂ rather than of an intermediate field?